src/gsy_e/gsy_e_core/util.py

```diff
@@ -443,21 +481,40 @@ def get_market_maker_rate_from_config(next_market, default_value=None, time_slot
                 raise e
         return find_object_of_same_weekday_and_time(GlobalConfig.market_maker_rate,
                                                     time_slot)
-    else:
-        return GlobalConfig.market_maker_rate
+    return GlobalConfig.market_maker_rate
+
+
+def get_feed_in_tariff_rate_from_config(next_market: "MarketBase",
+                                        default_value=None, time_slot=None):
+    """Get feed in tariff rate from config."""
+    if next_market is None:
+        return default_value
+    if isinstance(GlobalConfig.FEED_IN_TARIFF, dict):
+        if time_slot is None:
+            try:
+                time_slot = next_market.time_slot
+            except AttributeError as e:
+                logging.exception("time_slot parameter is required for future markets.")
+                raise e
+        return find_object_of_same_weekday_and_time(GlobalConfig.FEED_IN_TARIFF,
+                                                    time_slot) or 0.
```

Comment:
```suggestion
        time_slot = time_slot or next_market.time_slot
        if not time_slot:
            return 0
        return find_object_of_same_weekday_and_time(GlobalConfig.FEED_IN_TARIFF,
                                                    time_slot) or 0.
```

Reply:
Reason for my suggestion is that AFAIS, the exception that you are re-rasing  is not caught anywhere and would make the sim crash.

Reply:
Rather than completely skipping the exception to go un notice, I would at least leave it with with `logging.error`

Reply:
Ok, agreed. But this case should never happen. If the next_market is a future market (i know, not for now), the time_slot parameter should be given. So actually, it would be worth an assertion here (as it would be an implementation error).
Please add the future market handling in the follow up PR. Otherwise this will end up in another hard to find bug. Thanks!

Reply:
@hannesdiedrich interesting indeed, I am fairly certain that I did not add it just for preventive reasons, but that I stumbled upon it. I suspect though that when I did the first implementation of this feature the future markets were a work in progress, and it might be that time_slot was missing as an argument back then. 
Nevertheless, I agree to go with your suggestion and to make it more iron-clad with an assertion. Thanks for catching it!

Reply:
One problem with this suggestion is that at this point `time_slot = next_market.time_slot` it would have already raised an `AttributeError` exception and the next line where `time_slot ` is asserted won't be executed. SO, I propose it like following if we want it to be iron clad with an assertion

```
def get_feed_in_tariff_rate_from_config(next_market: "MarketBase", time_slot=None):
    """Get feed in tariff rate from config."""
    if next_market is None:
        return 0.
    if isinstance(GlobalConfig.FEED_IN_TARIFF, dict):
        if time_slot is None:
            try:
                time_slot = next_market.time_slot
            except AttributeError:
                assert False, "time_slot parameter is missing to get feed-in tariff"
        return find_object_of_same_weekday_and_time(GlobalConfig.FEED_IN_TARIFF,
                                                    time_slot) or 0.
    return GlobalConfig.FEED_IN_TARIFF
```

Reply:
@faizan2590 please have a look at my suggestion. It does not contain the try/except block. Also, I explained why the Assertion in you shared code will never be triggered.

Reply:
Sorry, yes overlooked this part ` If this is not provided AND the market does not have a time_slot member, then this Exception is raised.
However, as mentioned above, also the future market has a time_slot parameter (it is just set to None). So, the AttributeError will never be triggered.`

I will adapt as per your suggestion

Reply:
Adapted: https://github.com/gridsingularity/gsy-e/pull/1338/commits/7e6a6030b42388f904aeabb6f3e004382348fd45